subsys/net/lib/openthread/platform/radio.c

```diff
@@ -159,88 +184,130 @@ void platformRadioInit(void)
 		return;
 	}
 
+	k_work_q_start(&ot_work_q, ot_task_stack,
+		K_THREAD_STACK_SIZEOF(ot_task_stack), OT_WORKER_PRIORITY);
+
 	__ASSERT(radio_api->get_capabilities(radio_dev)
```

Comment:
Not part of your patch, but I just wonder if it is proper to put this kind of check to assert. If the asserts are not compiled in, the check just disappears and in this case it probably should not.

Reply:
IMO get_capabilities is not something that could return different values for same devices. If you confirmed it to be working on nrf52840 there is no point testing it on all nrf52840 produced devices each time they power on. (But you still need to test it on other nrf52xx family devices).

Reply:
I do not understand you comment, how is this assert related to running this line of code in different hw. My point was that if asserts are not compiled in, then the get_capabilities() check is not done at all.

Reply:
And that is the point. This check does not need to be done in the field. If it was confirmed to evaluate "true" on a platform(e.g. in auto tests) it will always be like that on this platform and can safely be compiled with asserts disabled. 
Isn't that the idea behind assert?

Reply:
Or did you mean that we do never compile with asserts on? I assumed that on this stage of development asserts are enabled by default:)

Reply:
Better to check always during power up than never at all. Changed to k_panic.

Reply:
If the api call is optional, then no issue. It just looked suspicious so I just wanted to clarify this.